src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -17437,5 +17431,498 @@ static void Main()
+
+            var compilation2 = CreateStandardCompilation(source2, new[] { compilation1.ToMetadataReference() }, options: TestOptions.DebugExe);
+
+            var expected = new[] {
+                // (2,18): error CS0122: 'I1.T1' is inaccessible due to its protection level
+                // class Test1 : I1.T1
+                Diagnostic(ErrorCode.ERR_BadAccess, "T1").WithArguments("I1.T1").WithLocation(2, 18),
+                // (6,12): error CS0122: 'I1.T1' is inaccessible due to its protection level
+                //         I1.T1 a = new Test1();
+                Diagnostic(ErrorCode.ERR_BadAccess, "T1").WithArguments("I1.T1").WithLocation(6, 12),
+                // (7,11): error CS0122: 'I1.T1.M1()' is inaccessible due to its protection level
+                //         a.M1();
+                Diagnostic(ErrorCode.ERR_BadAccess, "M1").WithArguments("I1.T1.M1()").WithLocation(7, 11),
+                // (8,41): error CS0122: 'I1.T2' is inaccessible due to its protection level
+                //         System.Console.WriteLine(new I1.T2());
+                Diagnostic(ErrorCode.ERR_BadAccess, "T2").WithArguments("I1.T2").WithLocation(8, 41),
+                // (9,41): error CS0122: 'I1.T3' is inaccessible due to its protection level
+                //         System.Console.WriteLine(new I1.T3());
+                Diagnostic(ErrorCode.ERR_BadAccess, "T3").WithArguments("I1.T3").WithLocation(9, 41),
+                // (10,37): error CS0122: 'I1.T4' is inaccessible due to its protection level
+                //         System.Console.WriteLine(I1.T4.B);
+                Diagnostic(ErrorCode.ERR_BadAccess, "T4").WithArguments("I1.T4").WithLocation(10, 37)
+                };
+            compilation2.VerifyDiagnostics(expected);
+
+            var compilation3 = CreateStandardCompilation(source2, new[] { compilation1.EmitToImageReference() }, options: TestOptions.DebugExe);
+
+            compilation3.VerifyDiagnostics(expected);
+        }
+
+        [Fact]
+        public void NestedTypes_07()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    private interface T1
+    {
+        void M1();
+    }
+
+    private class T2
+    {}
+
+    private struct T3
+    {}
+
+    private enum T4
+    {
+        B
+    }
+
+    class Test1 : I1.T1
+    {
+        static void Main()
+        {
+            I1.T1 a = new Test1();
+            a.M1();
+            System.Console.WriteLine(new I1.T2());
+            System.Console.WriteLine(new I1.T3());
+            System.Console.WriteLine(I1.T4.B);
+        }
+
+        public void M1()
+        {
+            System.Console.WriteLine(""M1"");
+        }
+    }
+}
+";
+            ValidateNestedTypes_01(source1, Accessibility.Private);
+        }
+
+        [Fact]
+        public void NestedTypes_08()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    private interface T1
+    {
+        void M1();
+    }
+
+    private class T2
```

Comment:
Though, it is theoretically possible, given the nature of the changes in this PR, I believe it is very unlikely to find any problem that doesn't exist with types that, let's say, are nested into classes, but exists for types nested into interfaces. Therefore, I see very little benefit in doing that kind of ad-hoc testing. At the same time, redoing full-matrix testing for nested types is cost-prohibitive. If you have reason to believe that there is a problem with some scenarios and that reason has a real basis behind it, I'd be happy to add tests for scenarios like this. Otherwise, I prefer to not perform discovery testing in context of this PR.